### DISCUSSION TITLE
What would be the best CUDA and Pytorch version for NequIP?

### DISCUSSION THREAD

**Discussion by turbosonics** [mir-group/nequip#367] [Q&A] [answered]
The place I'm working only has CUDA 11.2, and I'm trying to figure out which pytorch version should I need to ask, but it seems not easy... 

Meanwhile, could anyone recommend the best CUDA and Pytorch version for NequIP? Maybe I can ask system admins for these versions. I know we have PyTorch >= 1.8, !=1.9, <=1.11.* but I just hope to learn about the optimal environment for NequIP before I ask to admins. 

Thanks.

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @turbosonics ,

Thanks for your interest in our code!

We currently still recommend PyTorch 1.11, which works officially with CUDA 11.3 or 10.2. I think 11.4 should also work. (See https://pytorch.org/get-started/previous-versions/#linux-and-windows-11)

**Reply by lizhiqiang778** [reply]
Hello developers, this is Zhiqiang. I have a question about compatibility between GPU and Cuda version. As you recommended,  PyTorch 1.11 is a good choice, which works officially with CUDA 11.3 or 10.2. However, for RTX 4090(SM_8.9), cuda 11.8 may be needed for compatibility. So, it seems in a stuck and I'm very confused about this. Any suggestion is feasible. Thanks in advance.

**Reply by Linux-cpp-lisp** [reply]
Hi @lizhiqiang100 ,

Good question---I would recommend to try either PyTorch 1.13 or (likely better) the latest stable 2.*, and to watch carefully for this issue: https://github.com/mir-group/nequip/discussions/311.  In principle, on the `develop` branch, #311 should be resolved, but it remains unclear whether this is true universally.

Regardless of whether it works or fails, please post and let us know what happens!  I think most of our existing information is from A100/V100s, so it would be interesting to know whether the issue (or any other issue) occurs on 4090s.

**Reply by Linux-cpp-lisp** [reply]
I've heard from others now that newer PyTorch versions seem to work for this case, but please still report back with what you do or do not observe!

**Reply by apoletayev** [reply]
Just in case other folks are still looking for this, we have been using 2.2.0 and 2.4.1 with CUDA 12.2 and 12.4 for the last few months, and have not experienced any slowdowns on A100.

**Reply by Linux-cpp-lisp** [reply]
Thanks @apoletayev for confirming---this continues to be our experience as well that the recent 2.* versions are OK.
